ledger/testing/consensusRange.go

```diff
@@ -69,38 +80,34 @@ var consensusByNumber = []protocol.ConsensusVersion{
 // created and inserted in consensusByNumber. At that point, your feature is
 // probably active in that version. (If it's being held in vFuture, just
 // increment your `start`.)
-func TestConsensusRange(t *testing.T, start, stop int, test func(t *testing.T, ver int, cv protocol.ConsensusVersion)) {
+func TestConsensusRange(t *testing.T, start, stop int, test func(t *testing.T, ver int, cv protocol.ConsensusVersion, cfg config.Local)) {
 	if stop == 0 { // Treat 0 as "future"
 		stop = len(consensusByNumber) - 1
 	}
 	require.LessOrEqual(t, start, stop)
+	cfg := config.GetDefaultLocal()
 	for i := start; i <= stop; i++ {
-		var version string
-		if i == len(consensusByNumber)-1 {
-			version = "vFuture"
-		} else {
-			version = fmt.Sprintf("v%d", i)
-		}
+		version := versionStringFromIndex(i)
 		t.Run(fmt.Sprintf("cv=%s", version), func(t *testing.T) {
-			test(t, i, consensusByNumber[i])
+			test(t, i, consensusByNumber[i], cfg)
```

Comment:
add a new func rand8()
```
func rand16(t *testing.T) uint64 {
	var twobytes [2]byte
	_, err := rand.Read(twobytes[:])
	require.NoError(t, err)
	return binary.LittleEndian.Uint16(twobytes[:])
}
```
and turn int into a random test, and remove `cfg.DisableLedgerLRUCache = true` lines below. In this case we'll be able eventually to execute all test with cache and without.
```suggestion
		disable := rand16(t) % 2 == 0
		t.Run(fmt.Sprintf("cv=%s-nolru=%v", version, disable), func(t *testing.T) {
			cfg.DisableLedgerLRUCache = disable
			test(t, i, consensusByNumber[i], cfg)
```

Reply:
sounds cool, I get it. 

So I assume we are fine with the randomness in the test (rather than run twice with different settings?)
(or generally over the time test cover both side: activated cache and deactivated)

Reply:
we do not want to double execution time but want coverage so randomness should do the trick. Since we remember setting in a test name, one can debug it easeily as needed.

Reply:
get it, randomness should cover both branch.